nanoPU_sim.py

```diff
@@ -101,24 +103,14 @@ def compute_num_pkts(msg_len):
     return msg_len/Simulator.max_pkt_len if (msg_len % Simulator.max_pkt_len == 0) else msg_len/Simulator.max_pkt_len + 1
 
 def find_first_one(bitmap):
-    """Find the first set bit in the provided bitmap
+    """Find the first set bit in the provided bitmap.
+       NOTE: this may not work as expected when doing: find_first_one(~value) if value is all 1's
     """
     if bitmap == 0:
         return None
-    # assert bitmap > 0, "ERROR: bitmap must be positive"
-    # bits = bin(bitmap)[2:][::-1]
-    # return bits.find('1')
-
     # Prevent twos complement sign issue.
     return int(math.log(bitmap & -bitmap, 2))
 
-def priority_encoder(bitmap):
-    """Select an index based on the deployed priority selection policies
-    """
-    first_one = True
-    if first_one:
-        return find_first_one(bitmap)
-
```

Comment:
I had not deleted this function because different protocols (ie Homa) may need different priority encoding mechanism. Maybe I misunderstood the use of this function.

Reply:
We can add separate functions like `find_last_one` later if we need it.

Reply:
Makes sense